tests/Api/Parser/JsonParserTest.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+use Aws\Api\Parser\Exception\ParserException;
+use Aws\Api\Service;
+use Aws\AwsClient;
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit\Framework\TestCase;
+
+/*
+ * This class tests the PHP specific functionality of the JsonParser;
+ * cross-SDK testing is done in ComplianceTest.php
+ */
+
+/**
+ * @covers \Aws\Api\Parser\JsonRpcParser
+ * @covers \Aws\Api\Parser\JsonParser
+ */
+class JsonParserTest extends TestCase
+{
+    public function testUnixTimestampReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => 932169600]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $result = $handler($command)->wait()['timestamp']->__toString();
+
+        self::assertEquals("1999-07-17T00:00:00+00:00", $result);
+    }
+    public function testTextDateReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => "July 17th, 1999"]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $result = $handler($command)->wait()['timestamp']->__toString();
+        self::assertEquals("1999-07-17T00:00:00+00:00", $result);
+    }
+
+    public function testISOTimestampReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => "1999-07-17T00:00:00"]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $result = $handler($command)->wait()['timestamp']->__toString();
+        self::assertEquals("1999-07-17T00:00:00+00:00", $result);
+    }
+
+    public function testNegativeNumberReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => "-10000000"]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $result = $handler($command)->wait()['timestamp']->__toString();
+        self::assertEquals("1969-09-07T06:13:20+00:00", $result);
+    }
+
+    public function testStringReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => "this text is not a date"]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $this->setExpectedException(ParserException::class);
```

Comment:
PHPUnit has @expectedException and @expectedExceptionMessage that can be used to annotate tests. However, if you use a data provider, `setExpectedException` is likely the right approach, as that can take an input. Note that it's probably also wise to test for the exception message.